packages/cli-lib/lib/interpolation.js

```diff
@@ -0,0 +1,139 @@
+ * @param {object} interpolationData - The data to interpolate with
+ * @returns {string} - The interpolated string
+ * @example
+ * interpolation('Hello {{name}}', { name: 'World' })
+ * // 'Hello World'
+ */
+const interpolation = (stringValue, interpolationData) => {
+  const interpolationIdentifierRegEx = new RegExp(
+    `${delimiters.interpolation.start}(.*?)${delimiters.interpolation.end}`,
+    'g'
+  );
+  const replaceQueue = [];
+  let match;
+
+  // while & reduce necessary because RegExp.exec is stateful and only runs
+  // from beginning to end of string
+  while ((match = interpolationIdentifierRegEx.exec(stringValue)) != null) {
+    const { 0: matchedText, 1: rawIdentifier, index } = match;
+    const identifier = rawIdentifier.trim();
+
+    if (identifier && !isHelperIdentifier(identifier)) {
+      replaceQueue.unshift(
+        generateReplaceFn(matchedText, index, interpolationData[identifier])
+      );
+    }
+  }
+
+  const compiledString = replaceQueue.reduce(
+    (currentValue, replaceFn) => replaceFn(currentValue),
+    stringValue
+  );
+
+  return compiledString;
+};
+
+/**
+ * Compile a string using a specified helper function
+ * @param {string} stringValue - The string to modify
+ * @param {object} helperIdentifier - Helper name
+ * @param {function} helperFn - Helper function to call on string
+ * @returns {string} - The modified string
+ * @example
+ * compileHelper('White {{#yellow}}yellow{{/yellow}}', 'yellow', (string) => { chalk.reset.yellow(string) }))))
+ * // 'White yellow' (with 'yellow' colored yellow)
+ */
+const compileHelper = (stringValue, helperIdentifier, helperFn) => {
+  const helperIdentifierRegEx = new RegExp(
+    `${delimiters.interpolation.start}(${delimiters.helpers.start}${helperIdentifier})${delimiters.interpolation.end}(.*?)${delimiters.interpolation.start}(${delimiters.helpers.end}${helperIdentifier})${delimiters.interpolation.end}`,
+    'g'
+  );
+  const replaceQueue = [];
+  let match;
+
+  // while & reduce necessary because RegExp.exec is stateful and only runs
+  // from beginning to end of string
+  while ((match = helperIdentifierRegEx.exec(stringValue)) != null) {
+    const {
+      0: matchedText,
+      1: rawHelperIdentifierStart,
+      2: innerText,
+      index,
+    } = match;
+    const identifier = rawHelperIdentifierStart
+      .replace(delimiters.helpers.start, '')
+      .trim();
+
+    if (identifier && helperFn) {
+      replaceQueue.unshift(
+        generateReplaceFn(matchedText, index, helperFn(innerText))
+      );
+    }
+  }
+
+  const compiledString = replaceQueue.reduce(
+    (currentValue, replaceFn) => replaceFn(currentValue),
+    stringValue
+  );
+
+  return compiledString;
+};
+
+const compileHelpers = stringValue => {
+  return Object.keys(helpers).reduce((currentStringValue, helperIdentifier) => {
+    return compileHelper(
+      currentStringValue,
+      helperIdentifier,
+      helpers[helperIdentifier]
```

Comment:
not a blocker, but the `compileHelper` func also has access to `helpers` so there's no need to pass this in as an arg.

Reply:
True, this is my functional programming instincts kicking in here 😂

Reply:
I'm thinking leaving this one as-is because it doesn't hurt anything and makes the function more functional and self-contained.